docs/docs/ProgrammingGuide/run-on-k8s.md

```diff
@@ -0,0 +1,85 @@
+---
+# **Run BigDL on Kubernetes**
+
+The [Apache Spark on Kubernetes](https://apache-spark-on-k8s.github.io/userdocs/index.html) project enables
+native support for submitting Spark application to a kubernetes cluster. As a deep learning library for Apache
+Spark, BigDL can also run on Kubernetes by leveraging Spark on Kubernetes.
+
+---
+## **Prerequisites**
+
+1. You need to have a running Kubernetes cluster that support Spark on Kubernetes. See [here](https://apache-spark-on-k8s.github.io/userdocs/running-on-kubernetes.html#prerequisites)
+
+2. You need to spin up the **resource staging server** for dependency management. See [Dependency Management](https://apache-spark-on-k8s.github.io/userdocs/running-on-kubernetes.html#dependency-management) (This is optional if all your application dependencies are
+packaged into your own custom docker image or resides in remote locations like HDFS. See [Dependency Management Without The Resource Staging Server](https://apache-spark-on-k8s.github.io/userdocs/running-on-kubernetes.html#dependency-management-without-the-resource-staging-server))
+
+---
+## **Docker images**
+
+BigDL already published pre-built docker images that can be deployed into containers with pods.
+
+The images are as follows:
+
+|Component|Image|
+|---|---|
+|Spark Driver Image|intelanalytics/spark-driver:v2.2.0-kubernetes-0.5.0|
+|Spark Executor Image|intelanalytics/spark-executor:v2.2.0-kubernetes-0.5.0|
+|Spark Initialization Image|intelanalytics/spark-init:v2.2.0-kubernetes-0.5.0|
+|PySpark Driver Image|intelanalytics/spark-driver-py:v2.2.0-kubernetes-0.5.0|
+|PySpark Executor Image|intelanalytics/spark-executor-py:v2.2.0-kubernetes-0.5.0|
+
+You may also build your own customized images.
```

Comment:
Should we first publish those docker images to docker hub and add the Docker files to BigDL? @glorysdj

Reply:
i will publish docker images and the dockerfiles

Reply:
image published to docker hub.
dockerfiles are in PR 2025